### DISCUSSION TITLE
HTTP Compression?

### DISCUSSION THREAD

**Discussion by digitalml** [uNetworking/uWebSockets.js#671] [Q&A] [answered]
Love this library, thanks!

Been using uWebSockets for webScokets but now I'm also trying to implement GET and POST. Obviously it's pretty easy except the compression part. I need to compress several semi large responses from a GET (500k) and compression: uWS.SHARED_COMPRESSOR, or any of the other compression enumerations do nothing to the GET response. 

Does compression for an HTTP response require manual implementation?  Can I somehow use a gZip lib without resorting to use express or fastify? I can't find anything in the documentation on this.

Any help is appreciated. Thank you.

**Comment by Wyzix33** [comment, marked as answer]
You can use zlib, this is the way i used it:
```
const zlib = require('zlib');
```
and in your code you should check if the client accept encoding
```
const acceptEncoding = req.getHeader('accept-encoding');
let readStream = toArrayBuffer(fs.readFileSync(fileName));
 if (/\bgzip\b/.test(acceptEncoding)) {
     res.writeHeader('Content-Encoding', 'gzip');
     readStream = zlib.gzipSync(readStream);
    } else if (/\bdeflate\b/.test(acceptEncoding)) {
     res.writeHeader('Content-Encoding', 'deflate');
     readStream = zlib.deflateSync(readStream);
    }
    // else just original uncompressed 
    res.end(readStream);
```

**Reply by digitalml** [reply]
@Wyzix33 thank you! This worked perfectly.
